File: core/event_bus.js
const events = {};

function on(event, fn) {
  if (!events[event]) events[event] = [];
  events[event].push(fn);
}

function emit(event, data) {
  if (!events[event]) return;
  events[event].forEach(fn => fn(data));
}

module.exports = { on, emit };

File: omega_dkf_v1/core/event_bus.py
class EventBus:
    def __init__(self):
        self.subscribers = {}

    def subscribe(self, event, fn):
        self.subscribers.setdefault(event, []).append(fn)

    def emit(self, event, data):
        for fn in self.subscribers.get(event, []):
            fn(data)
